contracts/EntryPoint.sol

```diff
@@ -79,16 +80,62 @@ contract EntryPoint is StakeManager {
         require(success);
     }
 
+    //per aggregator information
+    struct AggregatorInfo {
+        // aggregator address
+        IAggregator aggregator;
+        // count of UserOps for this aggregator
+        uint count;
+        // aggregated signature
+        bytes signature;
+    }
+
+    //run all aggregators, each on its respective wallets
+    function validateAggregatedSignatures(
+        UserOperation[] calldata ops,
+        AggregatorInfo[] calldata aggregators
+    ) private view returns (uint totalCount) {
+        if (aggregators.length == 0) return 0;
+        uint startPtr = getMemoryPointer();
+        AggregatorInfo[] memory _aggregators = aggregators;
+
+        uint index = 0;
+        // uint loopPtr = getMemoryPointer();
+        for (uint i = 0; i < _aggregators.length; i++) {
+            // restoreMemoryPointer(loopPtr);
+            AggregatorInfo memory agg = _aggregators[i];
+            uint count = agg.count;
+            UserOperation[] memory copyOfOps = copyOpsRange(agg.aggregator, ops, index, index + count);
+            agg.aggregator.validateSignatures(address(this),
```

Comment:
How do you check that this 'aggregator' is indeed the one trusted by the wallet? I see that you use the wallet's own 'aggregator' in `simulateValidation` but here the 'aggregator' address is just a transaction input.